rdrf/registry/patients/admin_forms.py

```diff
@@ -739,27 +754,39 @@ def clean_working_groups(self):
             "disabled" in self.fields["working_groups"].widget.attrs
             or self.fields["working_groups"].disabled
         )
-        empty_choices = not self.fields["working_groups"].choices
+        base_working_group_choices = self.fields["working_groups"].choices
+
+        additional_working_group_fields = [
+            field_name
+            for field_name in self.data.keys()
+            if field_name.startswith("working_groups_")
+        ]
+        additional_working_group_values = [
+            value
+            for field_name in additional_working_group_fields
+            for value in self.data.getlist(field_name)
+        ]
 
         if is_disabled:
-            if empty_choices:
+            if not base_working_group_choices:
                 working_groups = WorkingGroup.objects.none()
             else:
-                working_groups = self.instance.working_groups.all()
+                unallocated_working_group = (
+                    WorkingGroup.objects.get_unallocated(self.registry_model)
+                )
+                if additional_working_group_values:
+                    working_groups = self.instance.working_groups.exclude(
+                        id=unallocated_working_group.id
+                    )
+                elif self.instance.working_groups.exists():
+                    working_groups = self.instance.working_groups.all()
+                else:
+                    working_groups = {unallocated_working_group}
```

Comment:
Can we do the following?

```suggestion
                    working_groups = unallocated_working_group
```

Using a `QuerySet` everywhere else, but a `set` here might cause issues later

Reply:
working_groups is expected to be a set of some type and a `union` operation is used before it returns. 

I could change it to be a queryset? e.g. `WorkingGroup.objects.filter(id=unallocated_working_group)`

Reply:
The other cases all have `working_groups` as a `QuerySet`, not a `set`, no? The union to create `all_selected_working_groups` works because `QuerySet` and `set` both have a `union` method, so this function will sometimes return a `QuerySet`, and sometimes a `set`, unless I'm misreading it?

Reply:
Also `WorkingGroup.objects.none()` is already a `QuerySet`

Reply:
Oh, I see, I was mixing up two of the cases :smile:.  Yes, your suggestion of `WorkingGroup.objects.filter(id=unallocated_working_group)` makes sense :+1: